nifi-nar-bundles/nifi-snmp-bundle/nifi-snmp-processors/src/main/java/org/apache/nifi/snmp/processors/AbstractSNMPProcessor.java

```diff
@@ -16,413 +16,248 @@
+    }
 
-    /** property to define host of the SNMP agent */
-    public static final PropertyDescriptor HOST = new PropertyDescriptor.Builder()
+    // SNMP versions
+    public static final AllowableValue SNMP_V1 = new AllowableValue("SNMPv1", "v1", "SNMP version 1");
+    public static final AllowableValue SNMP_V2C = new AllowableValue("SNMPv2c", "v2c", "SNMP version 2c");
+    public static final AllowableValue SNMP_V3 = new AllowableValue("SNMPv3", "v3", "SNMP version 3 with improved security");
+
+    // SNMPv3 privacy protocols
+    public static final AllowableValue NO_AUTH_NO_PRIV = new AllowableValue("noAuthNoPriv", "No authentication or encryption",
+            "No authentication or encryption.");
+    public static final AllowableValue AUTH_NO_PRIV = new AllowableValue("authNoPriv", "Authentication without encryption",
+            "Authentication without encryption.");
+    public static final AllowableValue AUTH_PRIV = new AllowableValue("authPriv", "Authentication and encryption",
+            "Authentication and encryption.");
+
+    // SNMPv3 authentication protocols
+    public static final AllowableValue MD5 = new AllowableValue("MD5", "MD5 based authentication",
+            "Provides authentication based on the HMAC-MD5 algorithm.");
+    public static final AllowableValue SHA = new AllowableValue("SHA", "SHA based authentication",
+            "Provides authentication based on the HMAC-SHA algorithm.");
+    public static final AllowableValue NO_AUTHENTICATION = new AllowableValue("", "None",
+            "Sends SNMP requests without authentication.");
+
+    // SNMPv3 encryption
+    public static final AllowableValue DES = new AllowableValue("DES", "DES",
+            "Symmetric-key algorithm for the encryption of digital data.");
+    public static final AllowableValue DES3 = new AllowableValue("3DES", "3DES",
+            "Symmetric-key block cipher, which applies the DES cipher algorithm three times to each data block.");
+
+    private static final String AES_DESCRIPTION = "AES is a symmetric algorithm which uses the same 128, 192, or 256 bit" +
+            " key for both encryption and decryption (the security of an AES system increases exponentially with key length).";
+
+    public static final AllowableValue AES128 = new AllowableValue("AES128", "AES128", AES_DESCRIPTION);
+    public static final AllowableValue AES192 = new AllowableValue("AES192", "AES192", AES_DESCRIPTION);
+    public static final AllowableValue AES256 = new AllowableValue("AES256", "AES256", AES_DESCRIPTION);
+    public static final AllowableValue NO_ENCRYPTION = new AllowableValue("", "None",
+            "Sends SNMP requests without encryption.");
+
+
+    public static final PropertyDescriptor AGENT_HOST = new PropertyDescriptor.Builder()
             .name("snmp-hostname")
-            .displayName("Host Name")
-            .description("Network address of SNMP Agent (e.g., localhost)")
+            .displayName("SNMP Agent Hostname")
```

Comment:
Localhost is allowed also, if we support hostnames we don't need a regex validator as it can be any string. Can we use something like InetAddress.getByName(String)? I can change the validator. If it successfully performs a DNS lookup, the hostname is valid, otherwise it isn't.

Reply:
As we discussed, domain name and ip validation together is complicated. It is up to the user to enter the correct domain or IP address, I'll replace this with a non empty validator and remove the network address validator from StandarValidators.